components/ProfilePage/ProfileButtons.tsx

```diff
@@ -16,45 +23,71 @@ export const StyledButton = styled(Button).attrs(props => ({
   }
 `
 
-export const ToggleProfilePrivate = styled(StyledButton).attrs(props => ({
-  className: `${
-    props.isProfilePublic ? "btn-secondary" : "btn-outline-secondary"
-  } col-12`
-}))`
-  ${props =>
-    props.isProfilePublic
-      ? `&:focus {
-    color: var(--bs-secondary);
-    background-color: white;
-    border-color: var(--bs-secondary);
-  }
-  &:hover {
-    color: white;
-    background-color: var(--bs-secondary);
-    border-color: white;
-  }`
-      : `background-color: white;   
-    &:focus{
-      color: white;
-      background-color:var(--bs-secondary);
-      border-color: white;
-    }
-    &:hover{
-      color: var(--bs-secondary);
-      background-color: white;
-      border-color: var(--bs-secondary);
-    }`}
-`
+type Props = {
+  formUpdated: boolean
+  uid: string
+  role: Role
+}
+
+export const ProfileEditToggle = ({ formUpdated, uid, role }: Props) => {
+  const { t } = useTranslation(["editProfile"])
+  return (
+    <FillButton
+      className={`py-1 ml-2 text-decoration-none`}
+      disabled={!!formUpdated}
+      href={`/ profile ? id = ${uid} `}
+      label={role !== "organization" ? t("viewMyProfile") : t("viewOrgProfile")}
+    />
+  )
+}
 
-export const EditProfileButton = () => {
+export const EditProfileButton = ({ className }: { className?: string }) => {
   const { t } = useTranslation("profile")
 
   return (
     <Internal
       href="/editprofile"
-      className="text-decoration-none text-white col-12"
+      className={`text-decoration-none text-white d-flex justify-content-center align-items-center col-12 ${className}`}
     >
-      <StyledButton>{t("button.editProfile")}</StyledButton>
+      <FillButton label={t("button.editProfile")} />
     </Internal>
   )
 }
+
+export function ProfileButtonsUser({
+  isProfilePublic,
+  onProfilePublicityChanged
+}: {
+  isProfilePublic: boolean | undefined
+  onProfilePublicityChanged: (isPublic: boolean) => void
+}) {
+  const { t } = useTranslation("editProfile")
+
+  const actions = useProfile()
+
+  const handleSave = async () => {
+    await updateProfile({ actions })
+  }
+  /** Only regular users are allowed to have a private profile. */
+  async function updateProfile({ actions }: { actions: ProfileHook }) {
+    const { updateIsPublic } = actions
+
+    await updateIsPublic(!isProfilePublic)
+    onProfilePublicityChanged(!isProfilePublic)
+  }
+  return (
+    <div className={`d-grid gap-2 col-12 m-3`}>
+      <EditProfileButton className={`py-1`} />
+      <ToggleButton
+        toggleState={isProfilePublic || false}
+        stateTrueLabel={t("forms.makePrivate")}
+        stateFalseLabel={t("forms.makePublic")}
+        onClick={handleSave}
+        className={`py-1`}
+      />
+    </div>
+  )
+}
+export function ProfileButtonsOrg({ isUser }: { isUser: boolean }) {
+  return <>{isUser ? <EditProfileButton /> : <FollowButton />}</>
```

Comment:
@sashamaryl It looks like we removed the feature flag check on the org Follow Button  - is that intentional?